samza-sql-shell/src/main/java/org/apache/samza/sql/client/cli/Main.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.sql.client.cli;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.samza.sql.client.impl.SamzaExecutor;
+import org.apache.samza.sql.client.interfaces.ExecutionContext;
+import org.apache.samza.sql.client.interfaces.SqlExecutor;
+import org.apache.samza.sql.client.util.CliException;
+import org.apache.samza.sql.client.util.CliUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class Main {
```

Comment:
I think we will have another Main.java in Li version. Could we take the common code out of this file so that it could be shared and keep this file lean ?

Reply:
I just realized we may not need to have another Main in Li version.  Currently, shell has made `SqlExecutor` as a plugin, so in Li version, we can just configure `shell.executor = Li.version.SamzaExecutor` in the configuration file `shell-defaults.xml`, then shell will use it just as the default executor.